### ISSUE TITLE
credentialSets does not support creation of 'UserAssigned' resource identity

### ISSUE THREAD

**Issue by tobiasholzner-whiteduck** [Azure/acr#790]
**Describe the bug**
I want to create a credentialSets resource with a UserAssigned identity using Bicep. 

Bicep template: https://learn.microsoft.com/en-us/azure/templates/microsoft.containerregistry/registries/credentialsets?pivots=deployment-language-bicep

Error:

`Resource type 'Microsoft.ContainerRegistry/registries/credentialSets' does not support creation of 'UserAssigned' resource identity. The supported types are 'SystemAssigned'`

**To Reproduce**

```
resource dockerHubCredentialSetResource 'Microsoft.ContainerRegistry/registries/credentialSets@2023-01-01-preview' = {
  name: 'dockerhub'
  parent: acrResource
  identity: {
    type: 'UserAssigned'
    userAssignedIdentities: {
      'XXXX': {}
    }
  }
  properties: {
    authCredentials: [
      {
        name: 'dockerhub'
        passwordSecretIdentifier: 'https://XXXX'
        usernameSecretIdentifier: 'https://XXXX'
      }
    ]
    loginServer: 'docker.io'
  }
}
```

**Expected behavior**
The resource gets created

**Comment by ChrisSidebotham** [comment]
@JXavierMSFT - Can you provide an update?

**Comment by JXavierMSFT** [comment]
Hey @ChrisSidebotham 

Thank you for submitting this feature request. We are evaluating this ask, I will update this thread when we have made a decision.

**Comment by toddysm** [comment]
Yes, this is in the works. We are looking to have ACR-to-ACR implementation with MIs available in the first half of 2026.

**Comment by toddysm** [comment]
UAMI are available for authentication with ACR registries upstream. See the following article: https://techcommunity.microsoft.com/blog/AzureCompute/use-azure-container-registry-as-an-upstream-source-for-artifact-cache/4517102

**Comment by audhage** [comment]
Why is this closed? The functionality linked by @toddysm doesn't cover this scenario. 
My understanding of the scenario is to attach user assigned managed identities to the `credentialSets` resource, not to the cache rule. 

For example, the credentialSets resource may need this to authenticate to a key vault to fetch the credentials to an upstream registry.
